force-app/main/default/classes/NextStepBatch.cls

```diff
@@ -20,38 +20,46 @@
  */
 public with sharing class NextStepBatch implements Database.Batchable<sObject>{
```

Comment:
Try using the stateful interface so that you can maintain state between executions
https://trailhead.salesforce.com/trailblazer-community/feed/0D54V00007T4Az1SAF